core/templates/services/user.service.ts

```diff
@@ -12,97 +12,126 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-
 /**
  * @fileoverview Service for user data.
  */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { AppConstants } from 'app.constants';
 import { UserInfo } from 'domain/user/user-info.model';
-require('services/contextual/url.service.ts');
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { UserBackendApiService } from 'services/user-backend-api.service';
+
+interface SubscriptionSummary {
+  'creator_picture_data_url': string;
+  'creator_username': string;
+  'creator_impact': number;
+}
+interface PreferencesBackendDict {
```

Comment:
We have already defined these in the backend API service, right?

Reply:
modified backend service to export these dict and referring them here.